HW0/P5/P5.txt

```diff
@@ -1 +1,21 @@
 # Your answers here.
+
+Problem 5
+   1. 128+64+32+16+8+4+2+1 = 255s
+   2. of course it will be faster
+      --> 32+4+1 = 37s ==> 6.89x faster
+   3. 8s ==> 31.875x faster
+   4. P5.py 
+   5. need to take one second to move numbers everytime after additions are done
+      8 + 7 = 15s
+   6. For this case, performance is limited by how many bags can be moved at
+      once, like how computing power is limited by memory bandwidth.
+      1) if cashier takes 1 bag?
+        all computations cannot be done until 256 bags are handed to cashiers.
+        After the last bag is moved to a cashier, the last addition can be 
+        made. So, I guess it will take 257s (256s + 1s)
```

Comment:
I think as long as taking bags from a cashier, coworkers can do computations while other coworkers haven't gotten bags yet. So, I guess the last computation will be done right after the last bag is taken to the last coworker.